infra/run-sollya.rkt

```diff
@@ -0,0 +1,267 @@
+#lang racket
+(require math/bigfloat
+         math/flonum)
+
+(define sollya-path (find-executable-path "sollya"))
+
+(provide sollya-compile sollya-apply sollya-kill)
+
+(struct sollya-machine
+  ([process #:mutable]
+   [out #:mutable]
+   [in #:mutable]
+   [err #:mutable]
+   buffer exprs vars prec backup))
+
+;----------------------------------------- INPUT PARSING ---------------------------------------------
+; Precision of the current program
+(define *precision* (make-parameter 53))
+
+(define function->sollya-format
+  (make-hash
+   `((pow . "(~a ^ ~a)")
+     (+ . "(~a + ~a)")
+     (- . "(~a - ~a)")
+     (/ . "(~a / ~a)")
+     (sqrt . "sqrt(~a)")
+     (* . "(~a * ~a)")
+     (fma . "((~a * ~a) + ~a)")                       ; shorthand for fma
+     (hypot . "sqrt(~a^2 + ~a^2)")                    ; shorthand for hypot
+     (exp . "exp(~a)")
+     (expm1 . "expm1(~a)")
+     (log . "log(~a)")
+     (log10 . "log10(~a)")
+     (log2 . "log2(~a)")
+     (log1p . "log1p(~a)")
+     ;(cbrt . "(~a^(1/3))")                           ; no cbrt
+     (sin . "sin(~a)")
+     (cos . "cos(~a)")
+     (tan . "tan(~a)")
+     (asin . "asin(~a)")
+     (acos . "acos(~a)")
+     (atan . "atan(~a)")
+     (sinh . "sinh(~a)")
+     (cosh . "cosh(~a)")
+     (tanh . "tanh(~a)")
+     (asinh . "asinh(~a)")
+     (acosh . "acosh(~a)")
+     (atanh . "atanh(~a)")
+     ;(atan2 . "atan(~a/~a)")                         ; no atan2
+     (erf . "erf(~a)")
+     (erfc . "erfc(~a)")
+     ;(tgamma . Gamma)                                ; no tgamma
+     ;(lgamma . LogGamma)                             ; no lgamma
+     (ceil . "ceil(~a)")
+     (floor . "floor(~a)")
+     (fmod . "mod(~a, ~a)")
+     ;(remainder . QuotientRemainer)                  ; no remainder
+     (fmax . "max(~a, ~a)")
+     (fmin . "min(~a, ~a)")
+     ;(truc . Truncate)                               ; no truncate
+     (round . "round(~a)")
+     (assert . "if (~a) then (~a) else nan")
```

Comment:
This may crash. When parsing `if` instead of `assert` I indeed parse it as `(if ...)`, but this thing crashes. We lose 2 benchmarks because of it.
I didn't handle `if` statement properly at this point